### ISSUE TITLE
Expose more internals: observable "subscriptionsCount" and access to "_subscriptions"

### ISSUE THREAD

**Issue by bago** [knockout/knockout#1875]
I'm the author of [Mosaico](https://github.com/voidlabs/mosaico), an opensource email template builder. Mosaico is developed around Knockout and I had to use some workaround in order to complete it.

I currently reference internal properties that are obfuscated in the production builds and this is weird.

1) I use [knockout-reactor](https://github.com/ZiadJ/knockoutjs-reactor) and reactor have [this code depending on the knockout obfuscation](https://github.com/ZiadJ/knockoutjs-reactor/blob/master/src/knockout.reactor.js#L188)
```javascript
    // Subscriptions are stored under either the _subscriptions field for the debug version
    // or the F, H or M fields when minified depending on the version used.
    var subscriptionsField;
    switch (ko.DEBUG || ko.version) {
        case true: subscriptionsField = '_subscriptions'; break;
        case "3.0.0": subscriptionsField = 'F'; break;
        case "3.1.0": subscriptionsField = 'H'; break;
        case "3.2.0": subscriptionsField = 'M'; break;
        case "3.3.0": subscriptionsField = 'G'; break;
        default: throw "Unsupported Knockout version. Only v3.0.0 to v3.3.0 are supported when minified. Current version is " + ko.version;
    }
```
The _subscriptions reference there is used to deal with subscriptions disposal when a "monitored" observable tree is removed from the "watch".

2) Mosaico direclty apply a similar hack in order to "observe" [the number of subscribers here](https://github.com/voidlabs/mosaico/blob/master/src/js/bindings/if-subs.js#L65)
```javascript
var beforeSubscriptionProp;
var afterSubscriptionProp;
if (typeof ko.DEBUG !== 'undefined' && typeof ko.isWritableObservable !== 'undefined') {
  beforeSubscriptionProp = 'beforeSubscriptionAdd';
  afterSubscriptionProp = 'afterSubscriptionRemove';
} else if (ko.version == "3.2.0") {
  beforeSubscriptionProp = 'va';
  afterSubscriptionProp = 'nb';
} else if (ko.version == "3.3.0") {
  beforeSubscriptionProp = 'ja';
  afterSubscriptionProp = 'ua';
}
else throw "Unsupported minimized Knockout version " + ko.version + " (supported DEBUG or minimized 3.2.0 ... 3.3.0)";
```
What I do In that "if-subs" binding is saying that a given block must be displayed only when the given observable has more than X subscribers (this is how Mosaico Editing tools in the Content/Style tabs are automagically hidden/shown depending on the visibility of the edited-content on the right pane).

My issue here is that getSubscriptionsCount is not an observable, so I have to create my own observable reacting to changes and to do that I replace the "private" beforeSubscriptionAdd/afterSubscriptionRemove methods with my own methods notifying its subscribers when the getSubscriptionsCount for the monitored observable change.

In summary: I wonder if I simply implemented my requirement the wrong ways and there are better supported ways to do that or if my needs are shared by other users and this means KO should expose more stuff in order to make these use-case "supported".

**Comment by brianmhunt** [comment]
I agree that being able to subscribe to changes in the subscription count (either as an observable or by events) would be useful.
